src/appHost.ts

```diff
@@ -589,12 +589,21 @@ miss: ${memoizedWithMissHit.miss}
         host.log.log('debug', `-- Removed slot keys: ${slotKeyToName(ownKey)} --`)
     }
 
-    function findDependantShells(declaringShell: PrivateShell): PrivateShell[] {
+    function findDependantShells(declaringShell: PrivateShell, dependantShells: Map<string, PrivateShell[]>): PrivateShell[] {
         return _([...addedShells.entries()])
             .flatMap(([name, shell]) => {
                 const dependencyAPIs = shell.entryPoint?.getDependencyAPIs?.() || []
                 const isDependant = dependencyAPIs.find(key => getAPIContributor(key)?.name === declaringShell.name)
-                return isDependant ? [shell, ...findDependantShells(shell)] : []
+                if (isDependant) {
+                    const value = dependantShells.get(name)
+                    if (value) {
+                        return value
+                    }
+                    const eps = [shell, ...findDependantShells(shell, dependantShells)]
+                    dependantShells.set(name, eps)
+                    return eps
+                }
+                return []
```

Comment:
I have 2 notes on this:
A. Let's rename `dependantShells` to `cache` for the sake of readability.
B. Let's avoid creating a side effect by mutating a map passed from outside function.
Instead - let's to the `set` inside the `flatMap` where we create the cache and treat here as read only